### PR TITLE
Use home page data from backend

### PR DESCRIPTION
**Pull request by codemonkey800**
## Description

#1036

- Moves code generating home sections data from frontend to mock server
- Refactor `hubAPI.getPluginSections()` to call the API directly

### CODE REVIEW

File: frontend/mock-server.js [hunk 1]
```diff
@@ -31,6 +31,98 @@ app.get('/plugins/index', async (_, res) => {
...
+
+    recently_updated: {
+      plugins: getSectionPlugins({
+        pluginMap,
+        limit,
+        sort: (a, b) => compareDates(a.release_date, b.release_date),
+      }),
+    },
+
+    top_installs: {
```
Comment (manasaV3) [inline frontend/mock-server.js:109]: This field would need to be updated.

**Review by manasaV3 — CHANGES_REQUESTED** [review]

File: frontend/mock-server.js (hunk 1 above)
Comment (richaagarwal) [inline frontend/mock-server.js:109]: Do these values need to match what's in the back-end? I believe this should be `top_installed` based on Manasa's latest changes in https://github.com/chanzuckerberg/napari-hub/pull/1132

File: frontend/mock-server.js [hunk 2]
```diff
@@ -31,6 +31,98 @@ app.get('/plugins/index', async (_, res) => {
...
+        new Set(
+          pluginIndex
+            .flatMap((plugin) => plugin.plugin_types || [])
+            .filter((pluginType) => pluginType !== 'theme'),
+        ),
+      ),
+    ) || 'sample_data';
+
+  const data = {
+    plugin_type: {
```
Comment (richaagarwal) [inline frontend/mock-server.js:84]: This is `plugin_types` in https://github.com/chanzuckerberg/napari-hub/pull/1132

**Review by manasaV3 — APPROVED** [review]
:shipit:

**Review by manasaV3 — DISMISSED** [review]
🚢

**Review by manasaV3 — APPROVED** [review]

**Pull request merged**